### PR TITLE
add unit testing for feature.js

### PR DESCRIPTION
**Pull request by irina060981**
https://github.com/alpheios-project/data-models/issues/31
Add unit testing for feature.js (according to the purpose - to increase overal unit testing cover)

### CODE REVIEW

File: tests/feature.test.js [hunk 1]
```diff
@@ -1,79 +1,286 @@
...
+
+    expect(feature.allowsUnrestrictedValues).toBeFalsy()
+    expect(feature.valuesUnrestricted).toBeFalsy()
   })
 
-  it('expects isEqual to correctly not equate two single value features of different type', () => {
-    let a = new Feature(Feature.types.note, 'valuea', Constants.LANG_LATIN)
-    let b = new Feature(Feature.types.frequency, 'valuea', Constants.LANG_LATIN)
-    expect(a.isEqual(b)).toBeFalsy()
+  it('4 Feature - check dataValuesFromInput, sort, compareTo, items, isEqual', () => {
```
Comment (balmas) [inline tests/feature.test.js]: I'm not sure if all of the equality and sorting tests that were included in the prior version are covered now. Is there a reason to remove them?

Comment (irina060981) [reply tests/feature.test.js]: I have changed the way of creating and going through the tests to increase coverage and it is more simple to create them from scratch.

And I believe this approach allows to track dependent changes easier.
For example - for comparing I have used isEqual and CompareTo in one unit case on the same features. And it goes through all code branches and checks.

I could add your test cases to mine - if you doubt of the resulted quality - no problems with it :)

Comment (balmas) [reply tests/feature.test.js]: Thank you :-) I'm not positive they weren't covered, just didn't immediately see how they were.

Comment (irina060981) [reply tests/feature.test.js]: I have looked more attentively on your tests and they are testing some cases that cover the same code from one point of view. But as for the future development and for future code changes - they will be really important.
So you were right - my tests were not as detailed as yours and they didn't cover all these variants listed in your tests.
So I have added them with a pleasure.
Thank you for being so attentive!

**Pull request merged**